### ISSUE TITLE
Why is output-model required?

### ISSUE THREAD

**Issue by greggroth** [isteven/angular-multi-select#276]
In my testing, the input-model is mutated, so that is sufficient for monitoring changes.  Why is `output-model` a required attribute?  Am I missing something?

**Comment by isteven** [comment]
Hi @greggroth , 

Well, first of all, the output-model consist only of the selected items (say, data.selected == true).

Second, you are correct. The input-model is mutated. You can indeed safely monitor & grab any changes there. 

However, the directive creates two extra properties into the input-model object for the purpose of grouping and indexing. You'll see these as "idx_xxxxx" and "spc_xxxxxx". The input-model is now "polluted", if I may say.

Now if you get output from the input-model manually, these won't be a problem. You can simply exclude them in your processing loop or such. However, it seems that some prefer to right away grab and save everything (say, using JSON.stringify()) - this is problematic because the extra properties will be included as well. 

This is where they use the output-model, in which the extra properties are removed. 

In short, output-model is the clean, selected items only.
